{"cells":[{"cell_type":"code","execution_count":1,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":24682,"status":"ok","timestamp":1691767136866,"user":{"displayName":"中田光隆","userId":"06690068059315868411"},"user_tz":-540},"id":"1W19Fsfntdvu","outputId":"e76bd3fd-63cf-402f-ba4f-9fe42f4f7b97"},"outputs":[{"output_type":"stream","name":"stdout","text":["Mounted at /content/drive\n"]}],"source":["from google.colab import drive\n","drive.mount('/content/drive')"]},{"cell_type":"code","execution_count":2,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":10,"status":"ok","timestamp":1691767136866,"user":{"displayName":"中田光隆","userId":"06690068059315868411"},"user_tz":-540},"id":"aK8OTJkqtdvx","outputId":"e296ee80-67a0-45f8-ab35-a52c41eb0dc2"},"outputs":[{"output_type":"stream","name":"stdout","text":["/content/drive/MyDrive/Colab Notebooks/pl_metric_learning_defect\n"]}],"source":["import os\n","os.chdir('/content/drive/MyDrive/Colab Notebooks/pl_metric_learning_defect')\n","\n","!pwd"]},{"cell_type":"code","execution_count":3,"metadata":{"executionInfo":{"elapsed":20968,"status":"ok","timestamp":1691767157831,"user":{"displayName":"中田光隆","userId":"06690068059315868411"},"user_tz":-540},"id":"9xkTH67qtdvy"},"outputs":[],"source":["!pip install -q timm wandb torchmetrics==0.11.1 pytorch-lightning==1.9.0 &> /dev/null\n","!pip install -q hydra-core --upgrade &> /dev/null"]},{"cell_type":"code","execution_count":4,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":10,"status":"ok","timestamp":1691767157831,"user":{"displayName":"中田光隆","userId":"06690068059315868411"},"user_tz":-540},"id":"EghHCTxHtdvz","outputId":"4c4b5fff-a0d5-4065-a335-f7c0b7453366"},"outputs":[{"output_type":"stream","name":"stdout","text":["'1. train.ipynb'\t config         model\t\t src\n","'2. sanity_test.ipynb'\t inference.py   output\t\t submit\n","'3. inference.ipynb'\t input\t        sanity_test.py\t train.py\n"]}],"source":["!ls"]},{"cell_type":"code","execution_count":6,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"2D1MAn3Ltdvz","outputId":"80f37b89-c5f3-46c3-93d1-f0800f01064e","executionInfo":{"status":"ok","timestamp":1691767721169,"user_tz":-540,"elapsed":37505,"user":{"displayName":"中田光隆","userId":"06690068059315868411"}}},"outputs":[{"output_type":"stream","name":"stdout","text":["\rSanity Checking: 0it [00:00, ?it/s]\rSanity Checking:   0% 0/2 [00:00<?, ?it/s]\rSanity Checking DataLoader 0:   0% 0/2 [00:00<?, ?it/s]/usr/local/lib/python3.10/dist-packages/pytorch_lightning/core/module.py:483: UserWarning: You called `self.log('val_loss', ..., logger=True)` but have no logger configured. You can enable one by doing `Trainer(logger=ALogger(...))`\n","  rank_zero_warn(\n","/usr/local/lib/python3.10/dist-packages/pytorch_lightning/core/module.py:483: UserWarning: You called `self.log('val_BinaryAccuracy', ..., logger=True)` but have no logger configured. You can enable one by doing `Trainer(logger=ALogger(...))`\n","  rank_zero_warn(\n","/usr/local/lib/python3.10/dist-packages/pytorch_lightning/core/module.py:483: UserWarning: You called `self.log('val_BinaryPrecision', ..., logger=True)` but have no logger configured. You can enable one by doing `Trainer(logger=ALogger(...))`\n","  rank_zero_warn(\n","/usr/local/lib/python3.10/dist-packages/pytorch_lightning/core/module.py:483: UserWarning: You called `self.log('val_BinaryRecall', ..., logger=True)` but have no logger configured. You can enable one by doing `Trainer(logger=ALogger(...))`\n","  rank_zero_warn(\n","/usr/local/lib/python3.10/dist-packages/pytorch_lightning/core/module.py:483: UserWarning: You called `self.log('val_BinarySpecificity', ..., logger=True)` but have no logger configured. You can enable one by doing `Trainer(logger=ALogger(...))`\n","  rank_zero_warn(\n","/usr/local/lib/python3.10/dist-packages/pytorch_lightning/core/module.py:483: UserWarning: You called `self.log('val_BinaryF1Score', ..., logger=True)` but have no logger configured. You can enable one by doing `Trainer(logger=ALogger(...))`\n","  rank_zero_warn(\n","/usr/local/lib/python3.10/dist-packages/pytorch_lightning/core/module.py:483: UserWarning: You called `self.log('val_BinaryFBetaScore', ..., logger=True)` but have no logger configured. You can enable one by doing `Trainer(logger=ALogger(...))`\n","  rank_zero_warn(\n","Epoch 0:   0% 0/63 [00:00<?, ?it/s] /usr/local/lib/python3.10/dist-packages/pytorch_lightning/core/module.py:483: UserWarning: You called `self.log('train_loss', ..., logger=True)` but have no logger configured. You can enable one by doing `Trainer(logger=ALogger(...))`\n","  rank_zero_warn(\n","/usr/local/lib/python3.10/dist-packages/pytorch_lightning/core/module.py:483: UserWarning: You called `self.log('train_BinaryAccuracy', ..., logger=True)` but have no logger configured. You can enable one by doing `Trainer(logger=ALogger(...))`\n","  rank_zero_warn(\n","/usr/local/lib/python3.10/dist-packages/pytorch_lightning/core/module.py:483: UserWarning: You called `self.log('train_BinaryPrecision', ..., logger=True)` but have no logger configured. You can enable one by doing `Trainer(logger=ALogger(...))`\n","  rank_zero_warn(\n","/usr/local/lib/python3.10/dist-packages/pytorch_lightning/core/module.py:483: UserWarning: You called `self.log('train_BinaryRecall', ..., logger=True)` but have no logger configured. You can enable one by doing `Trainer(logger=ALogger(...))`\n","  rank_zero_warn(\n","/usr/local/lib/python3.10/dist-packages/pytorch_lightning/core/module.py:483: UserWarning: You called `self.log('train_BinarySpecificity', ..., logger=True)` but have no logger configured. You can enable one by doing `Trainer(logger=ALogger(...))`\n","  rank_zero_warn(\n","/usr/local/lib/python3.10/dist-packages/pytorch_lightning/core/module.py:483: UserWarning: You called `self.log('train_BinaryF1Score', ..., logger=True)` but have no logger configured. You can enable one by doing `Trainer(logger=ALogger(...))`\n","  rank_zero_warn(\n","/usr/local/lib/python3.10/dist-packages/pytorch_lightning/core/module.py:483: UserWarning: You called `self.log('train_BinaryFBetaScore', ..., logger=True)` but have no logger configured. You can enable one by doing `Trainer(logger=ALogger(...))`\n","  rank_zero_warn(\n","Epoch 0:  63% 40/63 [00:08<00:04,  4.88it/s, loss=0.741]\n","Validation: 0it [00:00, ?it/s]\u001b[A\n","Validation:   0% 0/13 [00:00<?, ?it/s]\u001b[A\n","Epoch 0:  95% 60/63 [00:11<00:00,  5.35it/s, loss=0.741]\n","Epoch 0: 100% 63/63 [00:11<00:00,  5.53it/s, loss=0.72, val_loss=2.370, val_BinaryAccuracy=0.420, val_BinaryPrecision=0.408, val_BinaryRecall=1.000, val_BinarySpecificity=0.0333, val_BinaryF1Score=0.580, val_BinaryFBetaScore=0.580]\n","Epoch 0: 100% 63/63 [00:17<00:00,  3.60it/s, loss=0.72, val_loss=2.370, val_BinaryAccuracy=0.420, val_BinaryPrecision=0.408, val_BinaryRecall=1.000, val_BinarySpecificity=0.0333, val_BinaryF1Score=0.580, val_BinaryFBetaScore=0.580, train_loss=1.370, train_BinaryAccuracy=0.575, train_BinaryPrecision=0.468, train_BinaryRecall=0.463, train_BinarySpecificity=0.650, train_BinaryF1Score=0.465, train_BinaryFBetaScore=0.465]\n","Testing DataLoader 0: 100% 50/50 [00:02<00:00, 20.97it/s]\n","┏━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━┓\n","┃\u001b[1m \u001b[0m\u001b[1m       Test metric       \u001b[0m\u001b[1m \u001b[0m┃\u001b[1m \u001b[0m\u001b[1m      DataLoader 0       \u001b[0m\u001b[1m \u001b[0m┃\n","┡━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━┩\n","│\u001b[36m \u001b[0m\u001b[36m      test_accuracy      \u001b[0m\u001b[36m \u001b[0m│\u001b[35m \u001b[0m\u001b[35m          0.42           \u001b[0m\u001b[35m \u001b[0m│\n","│\u001b[36m \u001b[0m\u001b[36m        test_auc         \u001b[0m\u001b[36m \u001b[0m│\u001b[35m \u001b[0m\u001b[35m           1.0           \u001b[0m\u001b[35m \u001b[0m│\n","│\u001b[36m \u001b[0m\u001b[36m       test_fbeta        \u001b[0m\u001b[36m \u001b[0m│\u001b[35m \u001b[0m\u001b[35m   0.3221131369798972    \u001b[0m\u001b[35m \u001b[0m│\n","│\u001b[36m \u001b[0m\u001b[36m       test_kappa        \u001b[0m\u001b[36m \u001b[0m│\u001b[35m \u001b[0m\u001b[35m   0.02684563758389258   \u001b[0m\u001b[35m \u001b[0m│\n","│\u001b[36m \u001b[0m\u001b[36m     test_precision      \u001b[0m\u001b[36m \u001b[0m│\u001b[35m \u001b[0m\u001b[35m   0.7040816326530612    \u001b[0m\u001b[35m \u001b[0m│\n","│\u001b[36m \u001b[0m\u001b[36m       test_recall       \u001b[0m\u001b[36m \u001b[0m│\u001b[35m \u001b[0m\u001b[35m   0.5166666666666667    \u001b[0m\u001b[35m \u001b[0m│\n","│\u001b[36m \u001b[0m\u001b[36m     test_specifity      \u001b[0m\u001b[36m \u001b[0m│\u001b[35m \u001b[0m\u001b[35m   0.03333333507180214   \u001b[0m\u001b[35m \u001b[0m│\n","└───────────────────────────┴───────────────────────────┘\n"]}],"source":["!python sanity_test.py"]}],"metadata":{"accelerator":"GPU","colab":{"machine_shape":"hm","provenance":[]},"gpuClass":"standard","kernelspec":{"display_name":"base","language":"python","name":"python3"},"language_info":{"name":"python","version":"3.9.7"},"orig_nbformat":4,"vscode":{"interpreter":{"hash":"24c1640307f24a58a0fa2f094f9233c094312f72655a74c470e0955fca66f830"}}},"nbformat":4,"nbformat_minor":0}